pkg/resources/database.go

```diff
@@ -227,11 +220,25 @@ func createDatabaseFromReplica(d *schema.ResourceData, meta interface{}) error {
 func ReadDatabase(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
 	dbx := sqlx.NewDb(db, "snowflake")
-	database, err := snowflake.ListDatabase(dbx, d.Get("name").(string))
+	name := d.Id()
+
+	stmt := snowflake.NewDatabaseBuilder(name).Show()
+	row := snowflake.QueryRow(db, stmt)
+	_, err := snowflake.ScanDatabase(row)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			// If not found, mark resource to be removed from statefile during apply or refresh
+			log.Printf("[DEBUG] database (%s) not found", name)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("unable to scan row for SHOW DATABASES")
+	}
```

Comment:
Why do we need the SHOW query? I assume this is to make sure the database with that name actually exists, would be good to leave a comment to explain why this is here.